### PR TITLE
Added posibility refactor java classes to client side

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**

### CODE REVIEW

File: plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/service/RefactoringServiceClientImpl.java [hunk 1]
```diff
@@ -71,6 +74,35 @@ public void makeCall(AsyncCallback<String> callback) {
         });
     }
 
+    @Override
+    public Promise<RenameRefactoringSession> createRenameRefactoring(final CreateRenameRefactoring createRenameRefactoring) {
```
Comment (skabashnyuk) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/service/RefactoringServiceClientImpl.java]: javadoc

File: plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/service/RefactoringServiceClientImpl.java [hunk 2]
```diff
@@ -71,6 +74,35 @@ public void makeCall(AsyncCallback<String> callback) {
...
+                asyncRequestFactory.createPostRequest(url, createRenameRefactoring)
+                                   .header(ACCEPT, APPLICATION_JSON)
+                                   .header(CONTENT_TYPE, APPLICATION_JSON)
+                                   .send(newCallback(callback, unmarshallerFactory.newUnmarshaller(RenameRefactoringSession.class)));
+            }
+        });
+    }
+
+
+    @Override
```
Comment (skabashnyuk) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/service/RefactoringServiceClientImpl.java]: javadoc

File: plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/rename/JavaRefactoringRename.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+import static org.eclipse.che.ide.ext.java.shared.dto.refactoring.RefactoringStatus.INFO;
+import static org.eclipse.che.ide.ext.java.shared.dto.refactoring.RefactoringStatus.OK;
+import static org.eclipse.che.ide.ext.java.shared.dto.refactoring.RefactoringStatus.WARNING;
+
+/**
+ * Class for rename refactoring java classes
+ * @author Alexander Andrienko
+ */
+@Singleton
+public class JavaRefactoringRename {
```
Comment (skabashnyuk) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/rename/JavaRefactoringRename.java]: test?

Comment (vparfonov) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/rename/JavaRefactoringRename.java]: added

File: plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/rename/RenameRefactoringAction.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @Override
+    public void update(ActionEvent event) {
+        CurrentProject currentProject = appContext.getCurrentProject();
+        if (currentProject == null) {
+            event.getPresentation().setEnabled(false);
+            return;
+        }
+        boolean isJavaProject = currentProject.getRootProject().getAttributes().get("language").contains("java");
```
Comment (vparfonov) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/rename/RenameRefactoringAction.java]: Need think about multi-module project

**Pull request merged**